pkg/infra/tower.go

```diff
@@ -0,0 +1,121 @@
+package infra
+
+import (
+	"log"
+	"math/rand"
+
+	"github.com/SOMAS2021/SOMAS2021/pkg/messages"
+)
+
+type Tower struct {
+	currPlatFood    float64
+	maxPlatFood     float64
+	currPlatFloor   uint64
+	agentCount      int
+	agents          map[string]*Base
+	agentsPerFloor  int
+	ticksPerDay     int
+	missingAgents   map[int][]int // key: floor, value: types of missing agents
+	reshufflePeriod int
+	tickCounter     int
+}
+
+func NewTower(currPlatFood float64, currPlatFloor uint64, agentCount,
+	agentsPerFloor, ticksPerDay, reshufflePeriod int) *Tower {
+	t := &Tower{
+		currPlatFood:    currPlatFood,
+		maxPlatFood:     currPlatFood,
+		currPlatFloor:   currPlatFloor,
+		agentCount:      agentCount,
+		agents:          make(map[string]*Base),
+		agentsPerFloor:  agentsPerFloor,
+		ticksPerDay:     ticksPerDay,
+		missingAgents:   make(map[int][]int),
+		reshufflePeriod: reshufflePeriod,
+		tickCounter:     1,
+	}
+	return t
+}
+
+func (t *Tower) Tick() {
+	//logs
+	log.Printf("A log from the tower! Tick no: %d", t.tickCounter)
+	log.Printf("The food left on the platform = %f", t.currPlatFood)
+
+	//useful parameters
+	day := 24 * 60
+	numOfFloors := t.agentCount / int(t.agentsPerFloor)
```

Comment:
In the future, I think that the number of floors should be chosen at instantiation and independent of number of agents

Reply:
Can be parametrized 👍 will be changed in the future and can be PRed by agent teams